src/ripple/core/SQLInterface_shards.h

```diff
@@ -0,0 +1,204 @@
+
+#ifndef RIPPLE_CORE_SQLINTERGACE_SHARDS_H_INCLUDED
+#define RIPPLE_CORE_SQLINTERGACE_SHARDS_H_INCLUDED
+
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/app/misc/Manifest.h>
+#include <ripple/core/Config.h>
+#include <ripple/core/DatabaseCon.h>
+#include <ripple/core/SQLInterface.h>
+#include <ripple/overlay/PeerReservationTable.h>
+#include <ripple/peerfinder/impl/Store.h>
+#include <boost/filesystem.hpp>
+
+namespace ripple {
+
+std::tuple<bool, std::unique_ptr<DatabaseCon>, std::unique_ptr<DatabaseCon>>
+makeLedgerDBs(
+    Config const& config,
+    beast::Journal const& j,
+    bool setupFromConfig,
+    LedgerIndex shardIndex,
+    bool backendComplete,
+    boost::filesystem::path const& dir,
+    DatabaseCon::CheckpointerSetup const& checkpointerSetup);
+
+boost::optional<LedgerIndex>
+getMinLedgerSeq(soci::session& db, SQLInterface::TableType type);
+
+boost::optional<LedgerIndex>
+getMaxLedgerSeq(soci::session& db, SQLInterface::TableType type);
+
+void
+deleteByLedgerSeq(
+    soci::session& db,
+    SQLInterface::TableType type,
+    LedgerIndex ledgerSeq);
+
+void
+deleteBeforeLedgerSeq(
+    soci::session& db,
+    SQLInterface::TableType type,
+    LedgerIndex ledgerSeq);
+
+int
+getRows(soci::session& db, SQLInterface::TableType type);
+
+std::tuple<int, int, int>
+getRowsMinMax(soci::session& db, SQLInterface::TableType type);
+
+bool
+updateLedgerDBs(
+    soci::session& txdb,
+    soci::session& lgrdb,
+    std::shared_ptr<Ledger const> const& ledger,
+    std::uint32_t const index,
+    beast::Journal const j,
+    std::atomic<bool>& stop);
+
+bool
+saveValidatedLedger(
+    DatabaseCon& ldgDB,
+    DatabaseCon& txnDB,
+    Application& app,
+    std::shared_ptr<Ledger const> const& ledger,
+    bool current);
+
+bool
+loadLedgerInfoByIndex(
+    soci::session& db,
+    SQLInterface::SQLLedgerInfo& info,
+    beast::Journal const& j,
+    LedgerIndex ledgerSeq);
+
+bool
+loadLedgerInfoByIndexSorted(
+    soci::session& db,
+    SQLInterface::SQLLedgerInfo& info,
+    beast::Journal const& j,
+    bool ascendSort);
+
+bool
+loadLedgerInfoByIndexLimitedSorted(
+    soci::session& db,
+    SQLInterface::SQLLedgerInfo& info,
+    beast::Journal const& j,
+    LedgerIndex ledgerFirstIndex,
+    bool ascendSort);
+
+bool
+loadLedgerInfoByHash(
+    soci::session& db,
+    SQLInterface::SQLLedgerInfo& info,
+    beast::Journal const& j,
+    uint256 const& ledgerHash);
+
+uint256
+getHashByIndex(soci::session& db, LedgerIndex ledgerIndex);
+
+bool
+getHashesByIndex(
+    soci::session& db,
+    beast::Journal const& j,
+    LedgerIndex ledgerIndex,
+    uint256& ledgerHash,
+    uint256& parentHash);
+
+void
+getHashesByIndex(
+    soci::session& db,
+    beast::Journal const& j,
+    LedgerIndex minSeq,
+    LedgerIndex maxSeq,
+    std::map<LedgerIndex, std::pair<uint256, uint256>>& map);
+
+int
+loadTxHistory(
+    soci::session& db,
+    Application& app,
+    Json::Value& txs,
```

Comment:
`txs` is the result and it's buried in the middle of the params. If the interface can't be re-written to get rid of the out param, consider making it the first param.

Reply:
Done. Made as return value.